src/main/index.js

```diff
@@ -108,9 +111,20 @@ app.on('ready', () => {
     } catch (error) {
       event.reply('PROCESS_VIDEO_ERR', 'ERR OCCURRED!');
     }
-
   })
 
+  ipcMain.on('SAVE_FILE', async (event, arg) => {
+    const results = await packVideoFromFrames(arg);
+  });
 
 
+  ipcMain.on('NEW_FRAME', async (event, arg) => {
+    const img = nativeImage.createFromDataURL(arg.imgb64);
+    if (!fs.existsSync('out-frames')) {
+      fs.mkdirSync('out-frames');
+    }
+    // TODO Promisfy writeFile?
+    // TODO JPEG Quality?
+    fs.writeFileSync(`out-frames/out${arg.frameNum}.jpg`, img.toJPEG(100), () => {});
```

Comment:
AFAIK the `fs.writeFile()` should support promises https://stackoverflow.com/questions/31978347/fs-writefile-in-a-promise-asynchronous-synchronous-stuff